service/worker/indexer/indexer.go

```diff
@@ -58,13 +58,14 @@ type (
 	}
 	// Indexer used to consumer data from kafka then send to ElasticSearch
 	Indexer struct {
-		esIndexName string
-		consumer    messaging.Consumer
-		esProcessor ESProcessor
-		config      *Config
-		logger      log.Logger
-		scope       metrics.Scope
-		msgEncoder  codec.BinaryEncoder
+		esIndexName        string
+		consumer           messaging.Consumer
+		primaryProcessor   ESProcessor
+		secondaryProcessor ESProcessor
```

Comment:
Got it. How about using a slice instead? It's easier.
```suggestion
		processors   []ESProcessor
```

Reply:
Yes, also checked with chatGPT, seems slice is a better options to simplify logic and scalability